### PR TITLE
Feat/scopechange fixes

### PR DESCRIPTION
**Pull request by Gustav-Eikaas**
# Description

Please include a summary of the change and which issue is fixed.
List any external dependencies that are required for this change.

Completes: AB#FILL_IN_YOUR_ISSUE_ID

## Checklist:

-   [x] I have performed a self-review of my own code.
-   [ ] I have linked my DevOps task using the AB# tag.
-   [ ] My code is easy to read, and comments are added where needed.
-   [ ] My code is covered by tests.

### CODE REVIEW

File: src/Core/WorkSpace/src/Functions/DataOperations/queryCacheArgs.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { QueryClient, UseQueryResult } from 'react-query';
+
+export interface QueryCacheArgs {
+    queryClient: QueryClient;
+    key: string;
+    queryApi: UseQueryResult<any[] | undefined, unknown>;
```
Comment (Gustav-Eikaas) [inline src/Core/WorkSpace/src/Functions/DataOperations/queryCacheArgs.ts:6]: type

File: src/Core/WorkSpace/src/Hooks/usePrefetchQueries.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { DeepImmutableArray } from '@dbeining/react-atom';
+import { useEffect } from 'react';
+import { useQueryClient } from 'react-query';
+import { PrefetchQueriesOptions } from '../WorkSpaceApi/workspaceState';
+
+export function usePrefetchQueries(
+    prefetchQueriesOptions: PrefetchQueriesOptions[] & DeepImmutableArray<PrefetchQueriesOptions>
+): void {
```
Comment (Gustav-Eikaas) [inline src/Core/WorkSpace/src/Hooks/usePrefetchQueries.ts:8]: check querycontext

File: src/apps/ScopeChangeRequest/Api/ScopeChange/Queries/Request/getHistory.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { httpClient } from '../../../../../../Core/Client/Functions/HttpClient';
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Api/ScopeChange/Queries/Request/getHistory.ts:1]: this file shouldnt be here

File: src/apps/ScopeChangeRequest/Components/DetailView/Components/History/HistoryItem.tsx [hunk 4]
```diff
@@ -1,17 +1,13 @@
...
-    dateToDateTimeFormat,
-} from '../../../Workflow/Utils/dateFormatting';
+import { dateToDateTimeFormat } from '../../../Workflow/Utils/dateFormatting';
 
 interface HistoryItemProps {
     item: LogEntry;
 }
 
 export function HistoryItem({ item }: HistoryItemProps): JSX.Element {
-    const date = convertUtcToLocalDate(new Date(item.createdAtUtc));
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/DetailView/Components/History/HistoryItem.tsx:13]: sjekk med Kjetil

File: src/apps/ScopeChangeRequest/Components/Form/ScopeChangeRequestEditForm.tsx [hunk 5]
```diff
@@ -29,16 +39,26 @@ export const ScopeChangeRequestEditForm = ({
     close,
 }: ScopeChangeRequestEditFormProps): JSX.Element => {
     const [relatedObjects, setRelatedObjects] = useState<TypedSelectOption[]>([]);
+    // const queryClient = useQueryClient();
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Form/ScopeChangeRequestEditForm.tsx:42]: remove

File: src/apps/ScopeChangeRequest/Components/Form/ScopeChangeRequestEditForm.tsx [hunk 6]
```diff
@@ -192,70 +212,125 @@ function filterElementsByType(items: TypedSelectOption[], type: ProcoSysTypes |
...
+        queryKey: string[],
+        queryFn: () => Promise<T>,
+        options?: FetchQueryOptions<T, unknown, T, string[]> | undefined
+    ) => Promise<T>
 ) {
-    const relations: TypedSelectOption[] = [];
+    request.commissioningPackages.forEach((x) => {
+        addToQueryCache(referencesKeys.commPkg(x.procosysNumber), () =>
+            getCommPkgById(x.procosysId)
+        ).then((commPkg) =>
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Form/ScopeChangeRequestEditForm.tsx:236]: add all to list and patch when data comes back

File: src/apps/ScopeChangeRequest/Components/Workflow/Contributor/Components/Contributors.tsx [hunk 7]
```diff
@@ -7,34 +7,50 @@ import {
...
-    const { workflowKeys } = useScopechangeQueryKeyGen(request.id);
-    const { workflowKeys: workflowMutationKeys } = useScopechangeMutationKeyGen(request.id);
+    const { workflowKeys } = scopeChangeQueryKeys(request.id);
+    const { workflowKeys: workflowMutationKeys } = scopeChangeMutationKeys(request.id);
+
+    const { mutateAsync: removeContributorAsync } = useScopeChangeMutation(
+        request.id,
+        workflowMutationKeys.deleteContributorKey(step.id),
+        removeContributor,
+        {
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Workflow/Contributor/Components/Contributors.tsx:50]: Optimistic update

File: src/apps/ScopeChangeRequest/Hooks/React-Query/useQueryCacheLookup.ts [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+export function useQueryCacheLookup(): QueryCacheLookup {
+    const queryClient = useQueryClient();
+
+    /**
+     * Fetches data from the queryCache
+     * @param queryKey
+     * @returns
+     */
+    function getQueryData<T>(queryKey: string[]): T | undefined {
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Hooks/React-Query/useQueryCacheLookup.ts:24]: getQueryCacheData

**Review by jaysencpp — APPROVED** [review]

**Pull request merged**